lib/Server.js

```diff
@@ -1272,53 +1272,6 @@ class Server {
      * }
      */
     if (typeof options.proxy !== "undefined") {
-      // TODO remove in the next major release, only accept `Array`
-      if (!Array.isArray(options.proxy)) {
-        if (
-          Object.prototype.hasOwnProperty.call(options.proxy, "target") ||
-          Object.prototype.hasOwnProperty.call(options.proxy, "router")
-        ) {
-          /** @type {ProxyConfigArray} */
-          (options.proxy) = [/** @type {ProxyConfigMap} */ (options.proxy)];
-        } else {
-          /** @type {ProxyConfigArray} */
-          (options.proxy) = Object.keys(options.proxy).map(
-            /**
-             * @param {string} context
-             * @returns {HttpProxyMiddlewareOptions}
-             */
-            (context) => {
-              let proxyOptions;
-              // For backwards compatibility reasons.
-              const correctedContext = context
-                .replace(/^\*$/, "**")
-                .replace(/\/\*$/, "");
-
-              if (
-                typeof (
-                  /** @type {ProxyConfigMap} */ (options.proxy)[context]
-                ) === "string"
-              ) {
-                proxyOptions = {
-                  context: correctedContext,
-                  target:
-                    /** @type {ProxyConfigMap} */
-                    (options.proxy)[context],
-                };
-              } else {
-                proxyOptions = {
-                  // @ts-ignore
-                  .../** @type {ProxyConfigMap} */ (options.proxy)[context],
-                };
-                proxyOptions.context = correctedContext;
-              }
-
-              return proxyOptions;
-            }
-          );
-        }
-      }
```

Comment:
We will need to write migration guide on this, because it is still the popular syntax :disappointed:

Reply:
Yes. We will create a separate migration guide for v5.